### PR TITLE
`StoreKit2TransactionListener`: cancel task on `deinit`

### PR DESCRIPTION
**Pull request by NachoSoto**
Currently the `Task` was outliving `StoreKit2TransactionListener`.

### CODE REVIEW

**Review by aboedo — APPROVED** [review]
great catch! I wonder if there's a good way to add a test for this so we don't allow ourselves to mess it up again?

**Comment by NachoSoto** [comment]
Ha thanks for asking me to write a test... because it made me find a retain cycle here and possibly other places where we use this pattern

File: Purchases/Purchasing/StoreKit2/StoreKit2TransactionListener.swift [hunk 1]
```diff
@@ -21,21 +21,29 @@ protocol StoreKit2TransactionListenerDelegate: AnyObject {
...
     weak var delegate: StoreKit2TransactionListenerDelegate?
 
     init(delegate: StoreKit2TransactionListenerDelegate?) {
         self.delegate = delegate
     }
 
     func listenForTransactions() {
-        self.taskHandle = Task {
+        self.taskHandle?.cancel()
+        self.taskHandle = Task { [weak self] in
```
Comment (NachoSoto) [inline Purchases/Purchasing/StoreKit2/StoreKit2TransactionListener.swift:33]: @aboedo fixed 🙃

**Comment by NachoSoto** [comment]
I'm going to look at other places where we might have been bitten by this.

File: SwiftStyleGuide.swift [hunk 2]
```diff
@@ -184,6 +184,11 @@ guard 1 == 1,
 // use _ = Task<Void, Never> for task initialization, so that if a part of the
 // Task throws, the compiler forces us to handle the exception
 // More info here: https://rev.cat/catching-exceptions-from-tasks
```
Comment (NachoSoto) [inline SwiftStyleGuide.swift:186]: I added a few notes here so we don't fall into this again hopefully.

**Comment by NachoSoto** [comment]
Okay checked and everywhere we use `Task`, the `Task` has a limited lifespan so no retain cycle to worry about other than here.

File: SwiftStyleGuide.swift (hunk 2 above)
Comment (aboedo) [reply SwiftStyleGuide.swift:186]: great call

**Review by aboedo — APPROVED** [review]

**Pull request merged**